pkg/clients/validation.go

```diff
@@ -49,7 +53,8 @@ func (p *PayloadValidator) Validate(ctx context.Context, method, endpoint string
 	if p.isCreateClient(method, endpoint) || p.isUpdateClient(method, endpoint) {
 		clientRequest := new(client.OAuth2Client)
 		if err := json.Unmarshal(body, clientRequest); err != nil {
-			return ctx, nil, err
+			p.logger.Error("Json parsing error: ", err)
```

Comment:
are we using the `fmt.Errorf("failed to parse JSON body")` to bubble it up to the frontend?
if not can we drop the logger from here and simply enrich the error with `err`?

```suggestion
return ctx, nil, fmt.Errorf("failed to parse JSON body: %s", err)
```

Reply:
Yes, my problem was that previously the message was pretty useless for another service or for a user (it was something like `Unable to parse into Go struct`). I prefer this, but no problem with reverting it.

Reply:
not a biggie in the great scheme of things, do me a favour, add a `TODO` saying to review usage of logger and let s move on